src/framework/mod.rs

```diff
@@ -56,6 +62,23 @@ impl HttpApiClient {
             http_client: reqwest::Client::new(),
         }
     }
+
+    pub fn new_custom(
+        credentials: auth::Credentials,
+        config: HttpApiClientConfig,
+    ) -> HttpApiClient {
+        let http_client = reqwest::Client::builder()
+            .timeout(config.http_timeout)
+            .build()
+            .unwrap(); // Ok to unwrap; the only time this fails is if the native TLS backend cannot
```

Comment:
If this value is `Err`, we've basically got a choice about how to handle - unwrap immediately, or do something (e.g. print a nice error). If we use `unwrap()`, we're making that choice for the user. I suggest changing this to `?` and making the method return `Result/Option` instead. Because that way, if the user just wants to `unwrap()` they're free to do so. But if they want to handle it some other way, they can do that too. We should give the user the flexibility to do that :)

Reply:
True! My initial thought about using unwrap() was that, if we return a `Result`, we will diverge from the usage of `new()` above--how do we feel about that?

Reply:
I feel fine - if the programmer doesn't want to handle it, they're free to unwrap it. Optionals/results are so common in the stdlib and this crate that I feel anyone using this crate will need to be familiar with them anyway.

Reply:
Cool! Using Result now :)